src/Dropdown/index.tsx

```diff
@@ -1,134 +1,119 @@
+      bottom: ${bottom}px;
+      margin-bottom: 5px;
+    `}
   `}
 `
 
-type Vertical = 'top' | 'bottom'
-type Horizontal = 'left' | 'right'
-type Layout = [Vertical, Horizontal] | 'auto'
+type Vertical = 'to-top' | 'to-bottom'
+type Horizontal = 'to-left' | 'to-right'
+interface ILayout {
+  horizontal: Horizontal
+  vertical: Vertical,
+}
 interface IDropdownProps {
   className?: string
   button: React.ReactNode
   opened: boolean
-  layout: Layout
+  layout?: ILayout | 'auto'
 }
 
-const Dropdown: React.FC<IDropdownProps> = ({
-  className,
-  button,
-  children,
-  opened,
-  layout
-}) => {
-  const dropdownRef = useRef<HTMLDivElement>(null)
-  const getDropdownPosition = (layout: Layout): IDropdownContentPosition => {
-    if (dropdownRef.current) {
-      const drop = dropdownRef.current
-      const rect = drop.getBoundingClientRect()
-      const parentRect = drop.parentElement!.getBoundingClientRect()
 
-      let horizontal: Horizontal
-      let vertical: Vertical
+const Dropdown: React.FC<IDropdownProps> = ({ 
+  className, 
+  button, 
+  children, 
+  opened, 
+  layout = 'auto'
+}) => {
+  const ref = useRef<HTMLDivElement>(null)
+  const [position, setPosition] = useState<IPosition>({})
+
+  const getPosition = (layout: ILayout | 'auto') => {
+    if (layout !== 'auto' && position !== undefined) {
+      return position
+    } else if (ref.current) {
+      const rect = ref.current.getBoundingClientRect()
+      const parentRect = ref.current.parentElement!.getBoundingClientRect()
+      
+      const makePosition = ({ horizontal, vertical }: ILayout): IPosition => ({
+        [horizontal === 'to-left' ? 'right' : 'left']: 0,
+        [vertical === 'to-top' ? 'bottom' : 'top']: parentRect.height
+      })
 
       if (layout === 'auto') {
-        const closeToBottomBound = window.innerHeight - 20 < parentRect.bottom + rect.height
-        vertical = !closeToBottomBound ? 'top' : 'bottom'
-        const closeToRightBound = window.innerWidth - 20 < parentRect.left + rect.width
-        horizontal = !closeToRightBound ? 'left' : 'right'
+        const breakDistance = 20
+        const closeToBottomBound = parentRect.bottom + rect.height > window.innerHeight - breakDistance
+        const closeToRightBound = parentRect.left + rect.width > window.innerWidth - breakDistance 
+
+        return makePosition({
+          vertical: closeToBottomBound ? 'to-top' : 'to-bottom', 
+          horizontal: closeToRightBound ? 'to-left' : 'to-right'
+        })
       } else {
-        vertical = layout[0]
-        horizontal = layout[1]
-      }
-
-      return {
-        [horizontal]: 0,
-        [vertical]: `${parentRect.height}px`
+        return makePosition(layout)
       }
     }
-    return {
-      left: 0
-    }
+    return {}
   }
 
-  const [dropdownPosition, setDropdownPosition] = useState<IDropdownContentPosition>(getDropdownPosition(layout))
+  const handlePosition = () => setPosition(getPosition(layout))
```

Comment:
I think you can bring setPosition inside getPosition -> then you only have to return the makePosition returns and you can remove last return -> this way your not setting state on each resize event.